### DISCUSSION TITLE
useSession() and NextJS App router

### DISCUSSION THREAD

**Discussion by gagarin55** [nextauthjs/next-auth#9367] [Help]
Good day!

Could you please explain on this comment for `useSession()` hook (https://github.com/nextauthjs/next-auth/blob/f50393684cdb4e1a02cb92b47444b69f27d1787d/packages/next-auth/src/react.tsx#L116):
```
You will likely not need `useSession` if you are using the [Next.js App Router (`app/`)


```

How using useSession() is connected with App Router? And why I not need it ?

**Comment by n2k3** [comment]
I was wondering the same thing! The same is noted for the `<SessionProvider />` component:
https://github.com/nextauthjs/next-auth/blob/f50393684cdb4e1a02cb92b47444b69f27d1787d/packages/next-auth/src/react.tsx#L336

I suppose it originated from [this comment](https://github.com/nextauthjs/next-auth/issues/7760#issuecomment-1584527047).

In the [v5 migration guide](https://authjs.dev/guides/upgrade-to-v5?authentication-method=server-component#authenticating-server-side) of the new Auth.js docs it says client components still need `useSession()` hook, and therefor the `<SessionProvider />` component as well. If anybody knows why we will not likely either of these, I'd like to know how to do it differently.

Update:
I conclude we can ignore this for now, as the official [Next.js + Auth.js example](https://github.com/nextauthjs/next-auth/tree/main/apps/examples/nextjs#readme) (which is [deployed on Vercel](https://next-auth-example.vercel.app/) as well) is using both the [`<SessionProvider />`](https://github.com/nextauthjs/next-auth/blob/main/apps/examples/nextjs/app/client-example/page.tsx#L18) and the [`useSession()`](https://github.com/nextauthjs/next-auth/blob/main/apps/examples/nextjs/components/client-example.tsx#L45) hook.
